tests/integration/tests/conftest.py

```diff
@@ -99,6 +101,15 @@ def node_count(request) -> int:
     return int(node_count_arg)
 
 
+def snap_versions(request) -> Iterator[Optional[str]]:
+    """An endless iterable of snap versions for each node in the test."""
+    marking = ()
+    if snap_version_marker := request.node.get_closest_marker("snap_versions"):
+        marking, *_ = snap_version_marker.args
+    # endlessly repeat of the configured snap version after exhausting the marking
+    return itertools.chain(marking, itertools.repeat(None))
+
+
```

Comment:
Alrighty -- this method is simple in that is produces a steady stream of snap versions to use every time its invoked. 

just don't try to store its results into a finite sequence b/c it returns an infinite number of versions.